### DISCUSSION TITLE
CSF3 setup phase?

### DISCUSSION THREAD

**Discussion by markwoon** [storybookjs/storybook#17653] [Help]
I've been using CSF3 and have been really happy so far.  I've certainly been able to remove a fair bit of boilerplate.

One thing I'd like to see is a way to add a setup phase/function for each story.  My primary use case is to set up context for the component (e.g. mock fetch requests, etc).

Unless there's a way to do this already that I've missed?

**Comment by shilman** [comment]
have you tried loaders?

https://storybook.js.org/docs/react/writing-stories/loaders

**Reply by markwoon** [reply]
Unfortunately loaders don't seem to work.

I'm not actually retrieving data (the component performs the fetch directly).

I'm doing something like:

```
export const Story = {
  loaders: [
    () => {
      fetchMock.get(/\/some\/path/13, {
        status: 200,
        body: {success: true},
      });
    },
  ],
  args: {
    id: 13,
  },
};
```

The `fetchMock` call mocks the fetch that the component will request.

**Comment by yannbf** [comment]
Hey @markwoon what about using an addon that handles the mocking for you?

https://github.com/mswjs/msw-storybook-addon

https://github.com/nutboltu/storybook-addon-mock

https://github.com/HipsterBrown/storybook-mirage

**Reply by markwoon** [reply]
`storybook-addon-mock` looks promising, but I'd still prefer to see some initialization hook.

In my specific case, I'm using fetchMock in all my other tests and would prefer to be consistent.

I can just as easily do:
```
export const Story = {
  render: (args) => {
    fetchMock.get(/\/some\/path/13, {
      status: 200,
      body: {success: true},
    });
    return <Component {...args} />
  },
  args: {
    id: 13,
  },
};
```
